src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -4383,7 +4383,22 @@ public CanContinueBatchResult CanContinueBatch(in CanContinueBatchParameters par
                     parameters.QueryContext.Documents.DoNotReuse = true;
                     parameters.IndexingContext.DoNotReuse = true;
 
-                    if (parameters.Stats.MapAttempts >= Configuration.MinNumberOfMapAttemptsAfterWhichBatchWillBeCanceledIfRunningLowOnMemory)
+                    switch (parameters.WorkType)
+                    {
+                        case IndexingWorkType.Map:
+                            if (parameters.Stats.MapAttempts >= Configuration.MinNumberOfMapAttemptsAfterWhichBatchWillBeCanceledIfRunningLowOnMemory)
+                                canContinue = false;
+                            break;
+                        case IndexingWorkType.References:
+                            if (parameters.Count >= Configuration.MinNumberOfMapAttemptsAfterWhichBatchWillBeCanceledIfRunningLowOnMemory)
```

Comment:
Don't we have `References.Count`?

Reply:
Also what about `IndexingWorkType.Cleanup`?

Reply:
we have `_stats.MapReferenceAttempts` and `_stats.MapReferenceSuccesses` but it's not relevant, because we have this [optimization](https://github.com/ravendb/ravendb/blob/dbbcec0d8b0a90bfbfd3909079b882d8114c3d30/src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs#L203), which use `CanContinueReferenceBatch` but don't update any of stats (which is expected because it skip references), but `totalProcessedCount` updated in all cases so I use it for stop references batch.

Reply:
I will use the Count as well for `IndexingWorkType.Cleanup`